rocketpy/Function.py

```diff
@@ -2270,15 +2322,326 @@ def integral(self, a, b, numerical=False):
+
+        Parameters
+        ----------
+        val : int, float
+            The value of the output.
+        start : int, float
+            Initial guess of the output.
+        tol : int, float
+            Tolerance for termination.
+
+        Returns
+        -------
+        result : ndarray
+            The value of the input which gives the output closest to val.
+        """
+        return optimize.root(
+            lambda x: self.getValue(x) - val,
+            start,
+            tol=tol,
+        ).x[0]
+
+    def average(self, lower, upper):
+        """
+        Returns the average of the function.
+
+        Parameters
+        ----------
+        lower : float
+            Lower point of the region that the average will be calculated at.
+        upper : float
+            Upper point of the region that the average will be calculated at.
+
+        Returns
+        -------
+        result : float
+            The average of the function.
+        """
+        return self.integral(lower, upper) / (upper - lower)
+
+    def averageFunction(self, lower=None):
+        """
+        Returns a Function object representing the average of the Function object.
+
+        Parameters
+        ----------
+        lower : float
+            Lower limit of the new domain. Only required if the Function's source
+            is a callable instead of a list of points.
+
+        Returns
+        -------
+        result : Function
+            The average of the Function object.
+        """
+        if isinstance(self.source, np.ndarray):
+            if lower is None:
+                lower = self.source[0, 0]
+            upper = self.source[-1, 0]
+            xData = np.linspace(lower, upper, 100)
+            yData = np.zeros(100)
+            yData[0] = self.source[:, 1][0]
+            for i in range(1, 100):
+                yData[i] = self.average(lower, xData[i])
+            return Function(
+                np.concatenate(([xData], [yData])).transpose(),
+                inputs=self.__inputs__,
+                outputs=[o + " Average" for o in self.__outputs__],
+            )
+        else:
+            if lower is None:
+                lower = 0
+            return Function(
+                lambda x: self.average(lower, x),
+                inputs=self.__inputs__,
+                outputs=[o + " Average" for o in self.__outputs__],
+            )
+
+    def compose(self, func, extrapolate=False):
+        """
+        Returns a Function object which is the result of inputing a function into a function
+        (i.e. f(g(x))). The domain will become the domain of the input function and the range
+        will become the range of the original function.
+
+        Parameters
+        ----------
+        func : Function
+            The function to be inputed into the function.
+
+        Returns
+        -------
+        result : Function
+            The result of inputing the function into the function.
+        """
```

Comment:
Missing some docs here

Reply:
Should be fixed by 7847e45bdc4f09e40b1e5e13a1d2095dd6466f04.